libsolidity/analysis/TypeChecker.cpp

```diff
@@ -420,6 +420,10 @@ void TypeChecker::endVisit(InheritanceSpecifier const& _inheritance)
 	if (base->isLibrary())
 		m_errorReporter.typeError(_inheritance.location(), "Libraries cannot be inherited from.");
 
+	// Interface can have no constructors - no need to validate
+	if (base->contractKind() == ContractDefinition::ContractKind::Interface)
```

Comment:
Not fully sure where you want that assertion, but the `newExpressionType` below has it already and the typechecker checks for constructors not present in interfaces.